pypsa/plot.py

```diff
@@ -41,244 +46,232 @@
                 msg = "The passed axis is not a GeoAxesSubplot. You can "
                 "create one with: \nimport cartopy.crs as ccrs \n"
                 "fig, ax = plt.subplots("
                 'subplot_kw={"projection":ccrs.PlateCarree()})'
                 raise ValueError(msg)
+            else:
+                self.ax = ax
 
-        x_, y_, _ = ax.projection.transform_points(transform, x.values, y.values).T
-        x, y = pd.Series(x_, x.index), pd.Series(y_, y.index)
-
-        if color_geomap is not False:
-            draw_map_cartopy(ax, geomap, color_geomap)
-
-        if boundaries is not None:
-            ax.set_extent(boundaries, crs=transform)
-    elif ax is None:
-        ax = plt.gca()
-    elif hasattr(ax, "projection"):
-        raise ValueError("Axis is a geo axis, but `geomap` is set to False")
-    if not geomap and boundaries:
-        ax.axis(boundaries)
-
-    ax.set_aspect("equal")
-    ax.axis("off")
-    ax.set_title(title)
+            x_, y_, _ = self.ax.projection.transform_points(
+                transform, self.x.values, self.y.values
+            ).T
+            self.x, self.y = pd.Series(x_, self.x.index), pd.Series(y_, self.y.index)
 
-    # Plot buses:
+            if geomap_colors is not False:
+                self._add_geomap_features(geomap, geomap_colors)
 
-    if jitter is not None:
-        x = x + np.random.uniform(low=-jitter, high=jitter, size=len(x))
-        y = y + np.random.uniform(low=-jitter, high=jitter, size=len(y))
+            if self.boundaries is not None:
+                self.ax.set_extent(self.boundaries, crs=transform)
 
-    patches = []
-    if isinstance(bus_sizes, pd.Series) and isinstance(bus_sizes.index, pd.MultiIndex):
+            self.area_factor = get_projected_area_factor(self.ax, self.n.srid)
+        else:
+            if ax is None:
+                self.ax = plt.gca()
+            else:
+                self.ax = ax
+            self.ax.axis(self.boundaries)
+        self.ax.set_aspect("equal")
+        self.ax.axis("off")
+        self.ax.set_title(title)
+
+    def add_jitter(self, jitter):
+        """
+        Add random jitter to data.
+
+        Parameters
+        ----------
+        x : numpy.ndarray
+            X data to add jitter to.
+        y : numpy.ndarray
+            Y data to add jitter to.
+        jitter : float
+            The amount of jitter to add. Function adds a random number between -jitter and
+            jitter to each element in the data arrays.
+
+        Returns
+        -------
+        x_jittered : numpy.ndarray
+            X data with added jitter.
+        y_jittered : numpy.ndarray
+            Y data with added jitter.
+        """
+        self.x = self.x + np.random.uniform(low=-jitter, high=jitter, size=len(self.x))
+        self.y = self.y + np.random.uniform(low=-jitter, high=jitter, size=len(self.y))
+
+        return self.x, self.y
+
+    def get_multiindex_busses(
```

Comment:
```suggestion
    def get_multiindex_buses(
```
and perhaps name it `draw_...`